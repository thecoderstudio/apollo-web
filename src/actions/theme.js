function changeTheme(theme) {
  return {
    type: 'CHANGE_THEME',
    theme
  };
};

export { changeTheme };
