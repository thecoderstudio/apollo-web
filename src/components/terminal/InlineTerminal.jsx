import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Card from '../Card';
import Terminal from './Terminal';
import Button from '../buttons/Button';
import OutlinedButton from '../buttons/OutlinedButton';

const propTypes = {
  agent: PropTypes.object.isRequired
};

const Container = styled(Card)`
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 300px;
`;

const ControlBar = styled.div`
  display: flex;
  margin: 16px 0px;
  flex-direction: row-reverse;
`;

class InlineTerminal extends React.PureComponent {
  constructor(props) {
    super(props);
    this.renderContents = this.renderContents.bind(this);
    this.renderUnopenedTerminalState = this.renderUnopenedTerminalState.bind(this);
    this.openShell = this.openShell.bind(this);
    this.closeShell = this.closeShell.bind(this);
    this.state = {
      openedShell: props.openedShell
    };
  }

  componentDidUpdate(prevProps) {
    if (prevProps.openedShell !== this.props.openedShell) {
      this.setState({
        openedShell: this.props.openedShell
      });
    }
  }

  renderContents() {
    const connected = this.props.agent.connectionState == 'connected';
    if (connected && this.state.openedShell) {
      return <Terminal agent={this.props.agent} />;
    } else {
      return this.renderUnopenedTerminalState(connected);
    }
  }

  openShell() {
    this.setState({openedShell: true});
  }

  closeShell() {
    this.setState({openedShell: false});
  }

  renderUnopenedTerminalState(connected) {
    const connectionState = this.props.agent.connectionState;
    return (
      <div>
        <h2>{connectionState}</h2>
        {connected && <OutlinedButton onClick={this.openShell}>Open shell</OutlinedButton>}
      </div>
    );
  }

  render() {
    return (
      <div className={this.props.className}>
        <Container>
          {this.renderContents()}
        </Container>
        <ControlBar>
          <OutlinedButton onClick={this.closeShell}>Close shell</OutlinedButton>
        </ControlBar>
      </div>
    );
  }
}

InlineTerminal.propTypes = propTypes;

export default InlineTerminal;
