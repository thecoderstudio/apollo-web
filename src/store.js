import { createStore } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'
import storage from 'redux-persist/lib/storage';
import mapTransformer from './transformers/map';
import rootReducer from './reducers/root';

const persistConfig = {
  key: 'root',
  whitelist: ['auth', 'agent'],
  transforms: [mapTransformer({ whitelist: 'agent' })],
  storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
export const store = createStore(persistedReducer);
export const persistor = persistStore(store);
