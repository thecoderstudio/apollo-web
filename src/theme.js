export const darkTheme = {
  name: "dark",
  black: "#181818",
  lightBlack: "#1B1B1B",
  white: "#F8F8F8",
  primary: "#02A3EE",
  accent: "#4D02EE",
  connectedColor: "#27AE60",
  disconnectedColor: "#C0392B",
  connectingColor: "#F39C12",
  error: "#ED3E2A"
};

function getBrowserPreferredTheme() {
  return darkTheme;
}

export const browserPreferredTheme = getBrowserPreferredTheme();
