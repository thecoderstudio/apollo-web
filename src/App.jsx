import React from 'react';
import { connect } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import styled, { createGlobalStyle, ThemeProvider } from 'styled-components';
import ProtectedRoute from './components/ProtectedRoute';
import NavBar from './components/NavBar';
import Notifications from './components/notifications/Notifications';
import Dashboard from './pages/Dashboard';
import Welcome from './pages/Welcome';
import AgentDetail from './pages/AgentDetail';
import TerminalPage from './pages/TerminalPage';
import Admin from './pages/Admin';
import NotFound from './pages/NotFound';
import { darkTheme } from './theme';

const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=B612:wght@400;700&family=Roboto:wght@300;400&display=swap');

  html, body {
    margin: 0px;
    padding: 0px;
    height: 100%;
    width: 100%;
    font-weight: 500;
    color: ${(props) => props.theme.white};
    background-color: ${(props) => props.theme.black};
  }

  body {
    font-family: 'Roboto', sans-serif;
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: 'B612', sans-serif;
  }
`;

const Content = styled.div`
  height: 100%;
  width: 100%;
`;

function App(props) {
  return (
    <div>
      <ThemeProvider theme={darkTheme}>
        <BrowserRouter>
          <Content>
            {props.authenticated && props.hasChangedPassword && <NavBar />}
            <Switch>
              <ProtectedRoute exact path="/" component={Dashboard} fallbackComponent={Welcome} />
              <ProtectedRoute exact path="/agent/:agentId/shell" component={TerminalPage} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/agent/:agentId" component={AgentDetail} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/admin" component={Admin} fallbackComponent={NotFound} role="admin" />
              <Route component={NotFound} />
            </Switch>
            <Notifications />
            <GlobalStyle />
          </Content>
        </BrowserRouter>
      </ThemeProvider>
    </div>
  );
}

export default connect(
  (state) => ({
    authenticated: state.authenticated,
    hasChangedPassword: state.currentUser.hasChangedInitialPassword
  })
)(App);
