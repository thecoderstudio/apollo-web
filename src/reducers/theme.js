import { browserPreferredTheme } from '../theme';

export default function themeReducer(state = { theme: browserPreferredTheme }, action) {
  switch (action.type) {
    case 'CHANGE_THEME':
      return { theme: action.theme };
    default:
      return state;
  }
}
