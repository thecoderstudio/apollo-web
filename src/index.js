import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { render } from 'react-dom';
import App from './App';
import { store, persistor } from './store';

render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <App />
    </PersistGate>
  </Provider>,
  document.getElementById('app')
);
