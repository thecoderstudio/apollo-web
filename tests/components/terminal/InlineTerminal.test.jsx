import React from 'react';
import { ThemeProvider } from 'styled-components';
import WS from 'jest-websocket-mock';
import { mount } from 'enzyme';
import { darkTheme } from '../../../src/theme';
import InlineTerminal from '../../../src/components/terminal/InlineTerminal';

function getComponentTags(connectionState) {
  return (
    <div>
      <ThemeProvider theme={darkTheme}>
        <InlineTerminal
          agent={{
            id: "fakeid",
            name: "agentName",
            connectionState
          }} />
      </ThemeProvider>
    </div>
  );
}


describe("inline terminal", () => {
  const server = new WS(`ws://localhost:1234/agent/a2346886-83ba-442d-9fb7-d024c6274e22/shell`);

  afterEach(() => {
    WS.clean();
  });

  it("correctly renders connected", () => {
    const wrapper = mount(getComponentTags("connected"));
    expect(wrapper).toMatchSnapshot();

    wrapper.find('button').at(0).simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it("correctly renders disconnected", () => {
    const wrapper = mount(getComponentTags("disconnected"));
    expect(wrapper).toMatchSnapshot();
  });

  it("correctly renders connecting", () => {
    const wrapper = mount(getComponentTags("connecting"));
    expect(wrapper).toMatchSnapshot();
  });

  it("disconnects", () => {
    const wrapper = mount(getComponentTags("connected"));
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('i').at(0).simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
