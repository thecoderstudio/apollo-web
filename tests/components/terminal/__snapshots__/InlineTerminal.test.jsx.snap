// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inline terminal correctly renders connected 1`] = `
.c0 {
  box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1),0 4px 6px -2px rgba(0,0,0,0.05);
  padding: 40px;
  border-radius: 10px;
  background-color: #1B1B1B;
}

.c2 {
  max-height: 50px;
  padding: 12px;
  border: none;
  background-color: #4D02EE;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
}

.c2:hover {
  opacity: 0.90;
}

.c2:focus {
  outline: none;
}

.c2:active {
  opacity: 0.80;
}

.c3 {
  background-color: transparent;
  border: 1px solid #4D02EE;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 400px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 16px 0px 16px 16px;
  height: 24px;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

<div>
  <Component
    theme={
      Object {
        "accent": "#4D02EE",
        "black": "#181818",
        "connectedColor": "#27AE60",
        "connectingColor": "#F39C12",
        "darkGrey": "#3E3E3E",
        "darkWhite": "#F1F1F1",
        "disconnectedColor": "#C0392B",
        "error": "#ED3E2A",
        "green": "#32E67E",
        "inactive": "#363636",
        "lightBlack": "#1B1B1B",
        "name": "dark",
        "overlay": "rgba(37, 37, 37, 0.6)",
        "primary": "#02A3EE",
        "red": "#ED3E2A",
        "warning": "#ED852A",
        "white": "#F8F8F8",
      }
    }
  >
    <InlineTerminal
      agent={
        Object {
          "connectionState": "connected",
          "id": "fakeid",
          "name": "agentName",
        }
      }
    >
      <div>
        <Styled(styled.div)>
          <div
            className="c0 c1"
          >
            <div>
              <h2>
                connected
              </h2>
              <Styled(styled.button)
                onClick={[Function]}
              >
                <button
                  className="c2 c3"
                  onClick={[Function]}
                >
                  Open shell
                </button>
              </Styled(styled.button)>
            </div>
          </div>
        </Styled(styled.div)>
        <styled.div>
          <div
            className="c4"
          />
        </styled.div>
      </div>
    </InlineTerminal>
  </Component>
</div>
`;

exports[`inline terminal correctly renders connected 2`] = `
.c0 {
  box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1),0 4px 6px -2px rgba(0,0,0,0.05);
  padding: 40px;
  border-radius: 10px;
  background-color: #1B1B1B;
}

.c3 {
  height: 100%;
  width: 100%;
  padding: 20px;
}

.c4 {
  height: 100%;
  width: 100%;
}

.c4 .xterm-viewport {
  overflow: hidden !important;
}

.c6 {
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 400px;
}

.c2 {
  padding: 0px;
  height: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 16px 0px 16px 16px;
  height: 24px;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c7 {
  color: #ED3E2A;
}

<div>
  <Component
    theme={
      Object {
        "accent": "#4D02EE",
        "black": "#181818",
        "connectedColor": "#27AE60",
        "connectingColor": "#F39C12",
        "darkGrey": "#3E3E3E",
        "darkWhite": "#F1F1F1",
        "disconnectedColor": "#C0392B",
        "error": "#ED3E2A",
        "green": "#32E67E",
        "inactive": "#363636",
        "lightBlack": "#1B1B1B",
        "name": "dark",
        "overlay": "rgba(37, 37, 37, 0.6)",
        "primary": "#02A3EE",
        "red": "#ED3E2A",
        "warning": "#ED852A",
        "white": "#F8F8F8",
      }
    }
  >
    <InlineTerminal
      agent={
        Object {
          "connectionState": "connected",
          "id": "fakeid",
          "name": "agentName",
        }
      }
    >
      <div>
        <Styled(styled.div)>
          <div
            className="c0 c1"
          >
            <Styled(WithTheme(Terminal))
              agent={
                Object {
                  "connectionState": "connected",
                  "id": "fakeid",
                  "name": "agentName",
                }
              }
            >
              <WithTheme(Terminal)
                agent={
                  Object {
                    "connectionState": "connected",
                    "id": "fakeid",
                    "name": "agentName",
                  }
                }
                className="c2"
              >
                <Terminal
                  agent={
                    Object {
                      "connectionState": "connected",
                      "id": "fakeid",
                      "name": "agentName",
                    }
                  }
                  className="c2"
                  theme={
                    Object {
                      "accent": "#4D02EE",
                      "black": "#181818",
                      "connectedColor": "#27AE60",
                      "connectingColor": "#F39C12",
                      "darkGrey": "#3E3E3E",
                      "darkWhite": "#F1F1F1",
                      "disconnectedColor": "#C0392B",
                      "error": "#ED3E2A",
                      "green": "#32E67E",
                      "inactive": "#363636",
                      "lightBlack": "#1B1B1B",
                      "name": "dark",
                      "overlay": "rgba(37, 37, 37, 0.6)",
                      "primary": "#02A3EE",
                      "red": "#ED3E2A",
                      "warning": "#ED852A",
                      "white": "#F8F8F8",
                    }
                  }
                >
                  <styled.div
                    className="c2"
                  >
                    <div
                      className="c3 c2"
                    >
                      <styled.div
                        data-testid="terminal"
                      >
                        <div
                          className="c4"
                          data-testid="terminal"
                        />
                      </styled.div>
                    </div>
                  </styled.div>
                </Terminal>
              </WithTheme(Terminal)>
            </Styled(WithTheme(Terminal))>
          </div>
        </Styled(styled.div)>
        <styled.div>
          <div
            className="c5"
          >
            <Styled(styled.i)
              active={true}
              className="material-icons"
              onClick={[Function]}
            >
              <i
                className="c6 c7 material-icons"
                onClick={[Function]}
              >
                clear
              </i>
            </Styled(styled.i)>
          </div>
        </styled.div>
      </div>
    </InlineTerminal>
  </Component>
</div>
`;

exports[`inline terminal correctly renders connecting 1`] = `
.c0 {
  box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1),0 4px 6px -2px rgba(0,0,0,0.05);
  padding: 40px;
  border-radius: 10px;
  background-color: #1B1B1B;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 400px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 16px 0px 16px 16px;
  height: 24px;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

<div>
  <Component
    theme={
      Object {
        "accent": "#4D02EE",
        "black": "#181818",
        "connectedColor": "#27AE60",
        "connectingColor": "#F39C12",
        "darkGrey": "#3E3E3E",
        "darkWhite": "#F1F1F1",
        "disconnectedColor": "#C0392B",
        "error": "#ED3E2A",
        "green": "#32E67E",
        "inactive": "#363636",
        "lightBlack": "#1B1B1B",
        "name": "dark",
        "overlay": "rgba(37, 37, 37, 0.6)",
        "primary": "#02A3EE",
        "red": "#ED3E2A",
        "warning": "#ED852A",
        "white": "#F8F8F8",
      }
    }
  >
    <InlineTerminal
      agent={
        Object {
          "connectionState": "connecting",
          "id": "fakeid",
          "name": "agentName",
        }
      }
    >
      <div>
        <Styled(styled.div)>
          <div
            className="c0 c1"
          >
            <div>
              <h2>
                connecting
              </h2>
            </div>
          </div>
        </Styled(styled.div)>
        <styled.div>
          <div
            className="c2"
          />
        </styled.div>
      </div>
    </InlineTerminal>
  </Component>
</div>
`;

exports[`inline terminal correctly renders disconnected 1`] = `
.c0 {
  box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1),0 4px 6px -2px rgba(0,0,0,0.05);
  padding: 40px;
  border-radius: 10px;
  background-color: #1B1B1B;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 400px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 16px 0px 16px 16px;
  height: 24px;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

<div>
  <Component
    theme={
      Object {
        "accent": "#4D02EE",
        "black": "#181818",
        "connectedColor": "#27AE60",
        "connectingColor": "#F39C12",
        "darkGrey": "#3E3E3E",
        "darkWhite": "#F1F1F1",
        "disconnectedColor": "#C0392B",
        "error": "#ED3E2A",
        "green": "#32E67E",
        "inactive": "#363636",
        "lightBlack": "#1B1B1B",
        "name": "dark",
        "overlay": "rgba(37, 37, 37, 0.6)",
        "primary": "#02A3EE",
        "red": "#ED3E2A",
        "warning": "#ED852A",
        "white": "#F8F8F8",
      }
    }
  >
    <InlineTerminal
      agent={
        Object {
          "connectionState": "disconnected",
          "id": "fakeid",
          "name": "agentName",
        }
      }
    >
      <div>
        <Styled(styled.div)>
          <div
            className="c0 c1"
          >
            <div>
              <h2>
                disconnected
              </h2>
            </div>
          </div>
        </Styled(styled.div)>
        <styled.div>
          <div
            className="c2"
          />
        </styled.div>
      </div>
    </InlineTerminal>
  </Component>
</div>
`;

exports[`inline terminal disconnects 1`] = `
.c0 {
  box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1),0 4px 6px -2px rgba(0,0,0,0.05);
  padding: 40px;
  border-radius: 10px;
  background-color: #1B1B1B;
}

.c2 {
  max-height: 50px;
  padding: 12px;
  border: none;
  background-color: #4D02EE;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
}

.c2:hover {
  opacity: 0.90;
}

.c2:focus {
  outline: none;
}

.c2:active {
  opacity: 0.80;
}

.c3 {
  background-color: transparent;
  border: 1px solid #4D02EE;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 400px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 16px 0px 16px 16px;
  height: 24px;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

<div>
  <Component
    theme={
      Object {
        "accent": "#4D02EE",
        "black": "#181818",
        "connectedColor": "#27AE60",
        "connectingColor": "#F39C12",
        "darkGrey": "#3E3E3E",
        "darkWhite": "#F1F1F1",
        "disconnectedColor": "#C0392B",
        "error": "#ED3E2A",
        "green": "#32E67E",
        "inactive": "#363636",
        "lightBlack": "#1B1B1B",
        "name": "dark",
        "overlay": "rgba(37, 37, 37, 0.6)",
        "primary": "#02A3EE",
        "red": "#ED3E2A",
        "warning": "#ED852A",
        "white": "#F8F8F8",
      }
    }
  >
    <InlineTerminal
      agent={
        Object {
          "connectionState": "connected",
          "id": "fakeid",
          "name": "agentName",
        }
      }
    >
      <div>
        <Styled(styled.div)>
          <div
            className="c0 c1"
          >
            <div>
              <h2>
                connected
              </h2>
              <Styled(styled.button)
                onClick={[Function]}
              >
                <button
                  className="c2 c3"
                  onClick={[Function]}
                >
                  Open shell
                </button>
              </Styled(styled.button)>
            </div>
          </div>
        </Styled(styled.div)>
        <styled.div>
          <div
            className="c4"
          />
        </styled.div>
      </div>
    </InlineTerminal>
  </Component>
</div>
`;
